sgx/ias-types/src/parsed.rs

```diff
@@ -0,0 +1,392 @@
+        /// environment.
+        const REMOVE_FROM_CRL = 1 << 6;
+        /// X509-specific, indicates that the cert was revoked because a keyUsage-like attribute was
+        /// changed. Probably never used in our environment
+        const PRIVILEGE_WITHDRAWN = 1 << 7;
+        /// The CA delegated the ability to publish CRLs to another authority, but that authority was
+        /// compromised
+        const ATTRIBUTE_AUTHORITY_COMPROMISE = 1 << 8;
+    }
+}
+
+/// An enumeration of errors returned by IAS as part of the signed quote.
+///
+/// This is defined in the [IAS API v6, S4.2.1](https://software.intel.com/sites/default/files/managed/7e/3b/ias-api-spec.pdf).
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+#[derive(Clone, Debug, Display, Hash, Eq, Ord, PartialEq, PartialOrd)]
+pub enum QuoteError {
+    /// EPID signature of the ISV enclave QUOTE was invalid
+    SignatureInvalid,
+
+    /// The EPID group has been revoked. See RevocationCause
+    GroupRevoked(RevocationCause, PlatformInfo),
+
+    /// The EPID private key used to sign the QUOTE has been revoked by signature
+    SignatureRevoked,
+
+    /// The EPID private key used to sign the QUOTE has been directly revoked (not by signature)
+    KeyRevoked,
+
+    /// The SigRL used for the quote is out of date
+    SigrlVersionMismatch,
+
+    /// The TCB level of the SGX platform is out of date
+    GroupOutOfDate(Option<PseManifestResult>, PlatformInfo),
+
+    /// The enclave requires additional BIOS configuration
+    ConfigurationNeeded(Option<PseManifestResult>, PlatformInfo),
+
+    /// The enclave requires software mitigation
+    SwHardeningNeeded(Option<PseManifestResult>),
+
+    /// The enclave requires additional BIOS configuration and software mitigation
+    ConfigurationAndSwHardeningNeeded(Option<PseManifestResult>, PlatformInfo),
+
+    /// Unknown error: {0}
+    Other(String),
+}
+
+pub enum Error {
+    Json(JsonError),
+    Timestamp(ChronoError),
+    PlatformInfo(EncodingError),
+    Nonce(FromHexError),
+    Quote(ParseQuoteError),
+    PseManifest(FromHexError),
+    EpidPseudonym(DecodeError),
+}
+
+impl From<JsonError> for Error {
+    fn from(src: JsonError) -> Error {
+        Error::Json(src)
+    }
+}
+
+impl From<ParseQuoteError> for Error {
+    fn from(src: ParseQuoteError) -> Error {
+        Error::Quote(src)
+    }
+}
+
+/// The parsed Attestation Verification Report Data, parsed from Report.http_body JSON after
+/// signature and cert validation.
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+#[derive(Clone, Debug, PartialEq, PartialOrd)]
+pub struct Report {
+    /// A unqiue ID of this report.
+    pub id: String,
+
+    /// The timestamp this report was generated, as the duration since time::SystemTime::UNIX_EPOCH
+    // FIXME: It would be nice to parse this into something useful, but it's not usable at this time
+    pub timestamp: DateTime<Utc>,
```

Comment:
Why is `DateTime<Utc> not useful?

Reply:
It's part copy-pasta, part truth: *clients* may be able to take advantage of this, but inside the enclave there's no source of trusted time, so there's nothing to compare this against.